x-pack/plugins/security_solution/public/common/components/exceptions/helpers.tsx

```diff
@@ -397,6 +381,105 @@ export const entryHasListType = (
   return false;
 };
 
+/**
+ * Returns the value for `file.Ext.code_signature` which
+ * can be an object or array of objects
+ */
+export const getCodeSignatureValue = (
+  alertData: Ecs
+): Array<{ subjectName: string; trusted: string }> => {
+  const { file } = alertData;
+  const codeSignature = file && file.Ext && file.Ext.code_signature;
+
+  // Pre 7.10 file.Ext.code_signature was mistakenly populated as
+  // a single object with subject_name and trusted.
+  if (Array.isArray(codeSignature) && codeSignature.length > 0) {
+    return codeSignature.map((signature) => ({
+      subjectName: (signature.subject_name && signature.subject_name[0]) ?? '',
+      trusted: (signature.trusted && signature.trusted[0]) ?? '',
+    }));
+  } else {
+    const signature: CodeSignature | undefined = !Array.isArray(codeSignature)
+      ? codeSignature
+      : undefined;
+    const subjectName: string | undefined =
+      signature && signature.subject_name && signature.subject_name[0];
+    const trusted: string | undefined = signature && signature.trusted && signature.trusted[0];
```

Comment:
I am assuming you are wanting to pull out _just_ the first element of the array for subject_name and trusted? If you happen to _not_ have an array in `signature.subject_name` and let's say a string you are going to end up with the first character instead.

Reply:
So my thinking was that after the check on line 396, the expected mapping for `file.Ext.code_signature` is now `file.Ext.code_signature.subject_name: string[]` and `file.Ext.code_signature.trusted: string[]` (https://github.com/elastic/kibana/pull/77611/files/8037658523f11c7ee42f5b33c38832c62d979b3e#diff-814404a179e7ee01cacb74a55182e0e6R8). So I don't think that would be a situation we run into, though I could be missing something.